Source/DafnyCore/Options/CommonOptionBag.cs

```diff
@@ -151,12 +151,19 @@ public enum AssertionShowMode { None, Implicit, All }
 java - Compile to Java.
 py - Compile to Python.
 cpp - Compile to C++.
+rs - Compile to Rust.
 
 Note that the C++ backend has various limitations (see Docs/Compilation/Cpp.md). This includes lack of support for BigIntegers (aka int), most higher order functions, and advanced features like traits or co-inductive types.".TrimStart()
   ) {
     ArgumentHelpName = "language",
   };
 
+  public static readonly Option<string> TargetExecPath = new(new[] { "--target-executable", "-T" }, @"
```

Comment:
Thank you for this feedback! This is exactly what I was unsure about. For C and Rust , it's the compiler. For Python and Javascript, it's the interpreter. For Java though, there are three different executables that get called after translation before it gets run. I typically like to err on the side of explicitness. I agree about the ambiguity, it hasn't sat right with me at all lol.

- For Python and Javascript, it's an "interpreter"
- For C++ and Rust, it's a compiler
- For Java, it's a compiler and runtime
- For C#,  it's also a compiler and runtime, but only presented as a compiler.

Reply:
I think for Java you haven't updated the back-end, so the option has no effect. For C# the option is used to choose the binary that runs an already produced `.dll`, so I would consider it being used as a runtime.

To be frank, I think it's better not to have the `--target-executable` option. For a particular back-end, there are no particular tools expected to exist. Dafny might also have a native back-end that produces a binary.

I think what we potentially do want is to be able to tell Dafny where the locations of particular tools are, such as where is `dotnet`, where is the python interpreter, etc. Whether this should be done through Dafny CLI options, through environment variables, or through some sort of machine scoped Dafny configuration, I'm not sure.

Reply:
I think I can see your reasoning. I also agree with the concern for CLI bloat. I think what I have put forth so far might be a little off the mark. Additionally, I think I should split this PR to be more targeted, scope-wise. I will split out the CLI and associated code changes, leaving only the changes related to the `python` executable.

As far as the purpose of `dafny run` and `dafny build`, I think it may be helpful to clarify in the available documentation their purpose. That being said, I am an outsider for this repository and language, but I think the inclusion of the `--target` option leaves a hole between the possible and intended use cases without some kind of further modifications, in line with your 3rd paragraph. The hole is that I can run the code using a target language, but I have no control over how it actually gets run. This leaves a bit of a functional conundrum as the `--target` option already implies that the user wants _some_ control over _how_ things happen, but the only control the user really gets is "language", which most users who care about language would likely already be familiar with the footguns associated with not being able to target versions or other options of that language.

Again, I'm a bit of a newcomer, but I believe this warrants additional discussion that I am happy to contribute to and can open a new issue to facilitate that discussion. In the mean time, I will re-scope the changes in this PR to allow the initial issue to be resolved without the other can of worms I seem to have opened :P

Reply:
> Again, I'm a bit of a newcomer, but I believe this warrants additional discussion that I am happy to contribute to and can open a new issue to facilitate that discussion.

It does warrant it! Thank you for opening that issue.